### ISSUE TITLE
Issue with cookie sharing between Asp.net core 2.2 and Asp.net 4.7 web apps

### ISSUE THREAD

**Issue by BobVeenhuis** [BobVeenhuis/cookie_sharing#1]
We have two websites for which we need single signon capability.  The two sites are setup for cookie authentication per the below startup routines, and the work flow is such that login is accomplished on the net core 2.2 site and the net 4.7 site authenticates via the authentication cookie from the net core site.  We have verified that the cookie is being created and is visible to both sites, but the net 4.7 site is not able to authenticate because the cookie unprotect is failing in the authentication handler.  Specifically, the TicketDataFormat.Unprotect(cookie) line returns null, which we assume means that the net 4.7 app is unable to decrypt the cookie.  Could you tell us what might be causing that?

Versions:
Asp.net core 2.2
Asp.net 4.7.2
Microsoft Owin 4.0.0
Microsoft Owin.Security.Interop 2.1.2

protected override async Task<AuthenticationTicket> AuthenticateCoreAsync()
{
    AuthenticationTicket ticket = null;
    try
    {
	string cookie = Options.CookieManager.GetRequestCookie(Context, Options.CookieName);
	ticket = Options.TicketDataFormat.Unprotect(cookie);
	
	if (ticket == null)
	{
	    _logger.WriteWarning(@"Unprotect ticket failed");
	    return null;
	}
	...
    }...
}

.Net Core startup:

public void ConfigureServices(IServiceCollection oServices)
{
    Services.Configure<CookiePolicyOptions>(options =>
    {
	options.CheckConsentNeeded = context => true;
	options.MinimumSameSitePolicy = SameSiteMode.None;
	options.Secure = CookieSecurePolicy.Always;
    });
    
    Services.AddDataProtection()
	.PersistKeysToFileSystem(new DirectoryInfo("C:\\KeyRing"))
	.SetApplicationName("SharedApp");
	
    Services.AddAuthentication("Identity.Shared")
	.AddCookie("Identity.Shared", options =>
	{
	    options.Cookie.SecurePolicy = CookieSecurePolicy.Always;
	    options.Cookie.Name = ".AspNet.Cookie";
	    options.LoginPath = new PathString("/Identity/Authentication/Login");
	    options.LogoutPath = new PathString("/Identity/Authentication/Logout");
	    options.AccessDeniedPath = new PathString("/Identity/Authentication/AccessDenied");
	});
}

.Net 4.7 Startup

public void Configuration(IAppBuilder app)
{
    app.UseCookieAuthentication(new CookieAuthenticationOptions
    {
	AuthenticationType = "Identity.Shared",
	AuthenticationMode = AuthenticationMode.Active,
	CookieName = ".AspNet.Cookie",
	LoginPath = new PathString("/Login.aspx"),
	TicketDataFormat = new AspNetTicketDataFormat(
	new DataProtectorShim(
	    DataProtectionProvider.Create(new DirectoryInfo("C:\\KeyRing"), 
		(builder) => { builder.SetApplicationName("SharedApp"); })         
	    .CreateProtector(
                        "Microsoft.AspNetCore.Authentication.Cookies.CookieAuthenticationMiddleware",
                        "Identity.Shared",
                        "v2"))),
	CookieManager = new ChunkingCookieManager(),
    });

    app.SetDefaultSignInAsAuthenticationType("Identity.Shared");
}

**Comment by BobVeenhuis** [comment]
Here's a little bit of additional info.  I looked into the SecureDataFormat code that performs the TicketDataFormat.Unprotect, and I found the following sequence:

1.  byte[] protectedData = _encoder.Decode(protectedText);
2.  byte[] userData = _protector.Unprotect(protectedData);
3. TData model = _serializer.Deserialize(userData);

I then applied those same steps to the authentication cookie that is being received by the .net 4.7 app, and I was able to determine that there was no problem with the _encoder.Decode step, but the   _protector.Unprotect failed with an "invalid payload" error.  Unfortunately, no additional details though on the cause.  So if anyone has any ideas on what might cause that, it would be helpful.
